### PR TITLE
*: Create a reusable metricClient in util

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

Eliminate duplicate code for starting a goroutine that push metrics.

### What is changed and how it works?

Extract the common code to util and remove the original duplicate
implementation.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

File: pkg/util/p8s.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+// MetricClient manage the periodic push to the Prometheus Pushgateway.
+type MetricClient struct {
+	addr     string
+	interval time.Duration
+	registry *prometheus.Registry
+}
+
+// Start run a loop of pushing metrics to Prometheus Pushgateway.
+func (mc MetricClient) Start(ctx context.Context, grouping map[string]string) {
```
Comment (WangXiangUSTC) [inline pkg/util/p8s.go]: seems all the grouping is `"instance": nodeid`, how about use nodeid as arg

Comment (suzaku) [reply pkg/util/p8s.go]: @july2993 What's your opinion? Do we want more flexibility here or shorter code?

Comment (july2993) [reply pkg/util/p8s.go]: i prefer keep `grouping`

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**